### PR TITLE
refactor: Use DCRFrontCard instead of TrailType

### PR DESCRIPTION
**Pull request by AshCorr**
## What does this change?

Use `DCRFrontCard` type instead of `TrailType` in `Containers`, `FrontCard`, and others.

## Why?

Right now we're enhancing `FEFrontCard` into `DCRFrontCard` and then casting `DCRFrontCard` into `TrailType` and hoping that our properties match up. 

I think this can make some things a bit confusing, if you try and add a property to `TrailType` (like I've done in https://github.com/guardian/dotcom-rendering/pull/6998) you'll suddenly get a bunch of errors all over the place about not being able cast `DCRFrontCard` to `TrailType`.

Another reason it can make things confusing is because it can also make it a bit hard to figure out where a property is coming from, for example we were using the `isBooosted` property from `TrailType`, but the `DCRFrontCard` doesn't have this property, the only reason it worked is because isBoosted was marked as Optional and we returned isBoosted from `enhanceCards` even though the return type did not specify `isBoosted`.

A quick fix to avoid errors is to mark your new `TrailType` property as optional, however, we use also this type to represent the response from our Onwards API so when you mark a type as Optional when it isn't you create additional complexity and create a disconnect in the API contract between DCR and Frontend.

I propose that on the Fronts side of DCR we don't touch `TrailType` at all, and instead use our fronts specific type `DCRFrontCard`.

### CODE REVIEW

File: dotcom-rendering/src/types/front.ts [hunk 1]
```diff
@@ -248,7 +248,10 @@ export type DCRFrontCard = {
 	webPublicationDate?: string;
 	image?: string;
 	kickerText?: string;
+	supportingContent?: DCRSupportingContent[];
 	snapData?: DCRSnapType;
+	isBoosted?: boolean;
+	isCrossword?: boolean;
```
Comment (ioannakok) [inline dotcom-rendering/src/types/front.ts:254]: I am a big fan of this change! I've always found this confusing! I have a question regarding the props you chose to add to the `DCRFrontCard`. [`TrailType` was extending `BaseTrailType`](https://github.com/guardian/dotcom-rendering/blob/main/dotcom-rendering/src/types/trails.ts#L6-L38) and it had some props coming from its supertype I do not see here, e.g. `ageWarning`, `shortUrl`, `commentCount` etc. Is this intentional?

Another difference I see is `showQuotedHeadline` used to be optional in `TrailType` but not in `DCRFrontCard`.

Comment (ioannakok) [reply dotcom-rendering/src/types/front.ts:254]: Makes sense, thanks!

**Review by ioannakok — APPROVED** [review]

File: dotcom-rendering/src/web/lib/dynamicSlices.tsx [hunk 2]
```diff
@@ -41,7 +45,7 @@ export const Card50_Card50 = ({
 		<UL direction="row" padBottom={true}>
 			{cards50.map((trail, index) => (
 				<LI
-					key={trail.shortUrl}
```
[the remark this answers is not in the document]
Comment (OllysCoding) [reply dotcom-rendering/src/web/lib/dynamicSlices.tsx:44]: Does using index as key not end up throwing an eslint warning? I vaguely remember this being the case

[the remark this answers is not in the document]
Comment (AshCorr) [reply dotcom-rendering/src/web/lib/dynamicSlices.tsx:44]: Hm yes, I think you're right

**Pull request merged**